Update JobSuccessPolicy documentations for the beta graduation

tenzen-y opened the pull request:
<!--

 Hello!

 Remember to ADD A DESCRIPTION and delete this note before submitting
 your pull request. The description should explain what will change,
 and why.

 PLEASE title the FIRST commit appropriately, so that if you squash all
 your commits into one, the combined commit message makes sense.
 For overall help on editing and submitting pull requests, visit:
  https://kubernetes.io/docs/contribute/suggesting-improvements/

 Use the default base branch, “main”, if you're documenting existing
 features in the English localization.

 If you're working on a different localization (not English), see
 https://kubernetes.io/docs/contribute/new-content/overview/#choose-which-git-branch-to-use
 for advice.

 If you're documenting a feature that will be part of a future release, see
 https://kubernetes.io/docs/contribute/new-content/new-features/ for advice.

-->

This PR update is just the feature gate stage since there are no user-facing changes in the beta stage.

Part of: https://github.com/kubernetes/enhancements/issues/3998

Princesso commented:
Hello @tenzen-y :wave: please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before Tuesday July 16th 2024 18:00 PST. Thank you!

carlory commented:
/hold
until https://github.com/kubernetes/kubernetes/pull/126017 is merged.

Shubham82 commented:
FYI: The following upstream PR is closed https://github.com/kubernetes/kubernetes/pull/126017, Instead of that https://github.com/kubernetes/kubernetes/pull/126064 is opened for the same, but there is some talk to merge several PRs into one for this enhancement, Here is the comment for it: https://github.com/kubernetes/kubernetes/pull/126064#issuecomment-2232586393
Wait until all corresponding PRs are merged.

Princesso commented:
> FYI: The following upstream PR is closed [kubernetes/kubernetes#126017](https://github.com/kubernetes/kubernetes/pull/126017), Instead of that [kubernetes/kubernetes#126064](https://github.com/kubernetes/kubernetes/pull/126064) is opened for the same, but there is some talk to merge several PRs into one for this enhancement, Here is the comment for it: [kubernetes/kubernetes#126064 (comment)](https://github.com/kubernetes/kubernetes/pull/126064#issuecomment-2232586393) Wait until all corresponding PRs are merged.

Hello @Shubham82, I noticed that all the aforementioned upstream PRs are closed. Not of them got merged. Could you kindly let us know if other PRs were opened instead?

Shubham82 commented:
Hi @Princesso, Here is the corresponding PR(combining all PRs into one), and it is merged now:
https://github.com/kubernetes/kubernetes/pull/126067

Shubham82 commented:
removing the hold label, as the corresponding Upstream PR is merged.
/unhold

Shubham82 commented:
Thanks @tenzen-y for the PR, The changes look good to me.

/lgtm

hacktivist123 commented:
Hello @tenzen-y 👋! I'm reaching out from the Docs team. Just checking in as we approach Docs Freeze on Tuesday July 30th 18:00 PDT. This documentation appears to still be under review. To meet the Docs Freeze, this PR must have a technical review as well as lgtm and approve labels applied, without any unaddressed comments or concerns from SIG Docs. The status of this enhancement is marked as at risk for docs freeze. Thank you!

tenzen-y commented:
> Hello @tenzen-y 👋! I'm reaching out from the Docs team. Just checking in as we approach Docs Freeze on Tuesday July 30th 18:00 PDT. This documentation appears to still be under review. To meet the Docs Freeze, this PR must have a technical review as well as lgtm and approve labels applied, without any unaddressed comments or concerns from SIG Docs. The status of this enhancement is marked as at risk for docs freeze. Thank you!

Thank you for reminding!
/assign @atiratree @soltysh

atiratree commented on content/en/docs/reference/command-line-tools-reference/feature-gates/job-success-policy.md:
```diff
@@ -10,5 +10,9 @@ stages:
+    defaultValue: true
+    fromVersion: "1.31"
 ---
 Allow users to specify when a Job can be declared as succeeded based on the set of succeeded pods.
```
Maybe we  could change this part in the doc

```
The Job that meets the success policy gets the SuccessCriteriaMet condition
```

into 

```
The Job that meets the success policy gets the SuccessCriteriaMet condition with a SuccessPolicy reason.
```

since we changed the scope of the condition in this release.

tenzen-y replied:
Great point!
Thank you!

tenzen-y replied:
Done.

tenzen-y commented:
@atiratree I addressed your comments. PTAL, thanks.

tengqm commented:
/lgtm
/approve
